src/nfsv4mountscript.sh

```diff
@@ -0,0 +1,432 @@
+            chattr -i -f $stunnel_conf_file
+            rm $stunnel_conf_file
+            exit 1
+        fi
+
+        local mountmap_entry="$nfs_host;$stunnel_conf_file;$stunnel_log_file;$stunnel_pid_file;$checksumHash"
+        chattr -f -i $MOUNTMAPv4
+        echo "$mountmap_entry" >> $MOUNTMAPv4
+        if [ $? -ne 0 ]; then
+            chattr -f +i $MOUNTMAPv4
+            eecho "[$mountmap_entry] failed to add!"
+            chattr -i -f $stunnel_conf_file
+            rm $stunnel_conf_file
+            exit 1
+        fi
+        chattr -f +i $MOUNTMAPv4
+
+    else
+        # EntryExistinMountMap is true. That means stunnel_conf_file already exist for the storageaccount.
+
+        # Check if stunnel_pid_file exist for storageaccount
+        stunnel_pid_file=`cat $MOUNTMAPv4 | grep "stunnel_$storageaccount.pid" | cut -d ";" -f4`
+        if [ ! -f $stunnel_pid_file ]; then
+            eecho "[FATAL] '${stunnel_pid_file}' does not exist!"
+            exit 1
+        fi
+
+        is_stunnel_running=$(netstat -anp | grep stunnel | grep `cat $stunnel_pid_file`)
+        if [ -z "$is_stunnel_running" ]; then
+            vecho "stunnel is not running! Restarting the stunnel"
+
+            stunnel_status=$(stunnel $stunnel_conf_file 2>&1)
+            if [ -n "$stunnel_status" ]; then
+                is_binding_error=$(echo $stunnel_status | grep "$LOCALHOST:$available_port: Address already in use")
+                if [ -z "$is_binding_error" ]; then
+                    eecho "[FATAL] Not able to start stunnel process for '${stunnel_conf_file}'!"
+                    exit 1
+                else
+                    find_next_available_port_and_start_stunnel "$stunnel_conf_file"
+                    is_stunnel_running=$?
+                    if [ $is_stunnel_running -ne 0 ]; then
+                        eecho "Failed to get the next available port and start stunnel."
+                        exit 1
+                    fi
+                fi
+            fi
+        fi
+
+        available_port=$(cat $stunnel_conf_file | grep accept | cut -d: -f2)
+        vecho "Local Port to use: $available_port"
+    fi
+
+    mount_output=$(mount -t nfs -o "$MOUNT_OPTIONS,port=$available_port" "${LOCALHOST}:${nfs_dir}" "$mount_point" 2>&1)
+    mount_status=$?
+
+    if [ -n "$mount_output" ]; then
+        pecho "$mount_output"
+        vecho "Mount completed: ${LOCALHOST}:${nfs_dir} on $mount_point with port:${available_port}"
+    fi
+
+    if [ $mount_status -ne 0 ]; then
+        eecho "Mount failed!"
+        exit 1
+    fi
+}
+
+# Check if aznfswatchdogv4 service is running.
+if ! ensure_aznfswatchdog "aznfswatchdogv4"; then
+    exit 1
+fi
+
+vecho "nfs_host=[$nfs_host], nfs_dir=[$nfs_dir], mount_point=[$mount_point], options=[$OPTIONS], mount_options=[$MOUNT_OPTIONS]."
+
+# MOUNTMAPv4 file must have been created by aznfswatchdog service.
+if [ ! -f $MOUNTMAPv4 ]; then
+    touch $MOUNTMAPv4
```

Comment:
This must have been done when watchdog service was started, also common.sh tries to create this when it is loaded for mountscript file as well. Trying this again should not be useful, you can check if it's a systemd system as done for v3.